themes/default/content/blog/troubleshooting-builds-with-diagnostics/index.md

```diff
@@ -0,0 +1,154 @@
+
+```bash
+pulumi up -v=5
+```
+
+How do you decide which level to ask for? In the classic style of professionals discussing complex data, "it depends." We often encourage people to start at level 9, which gives you almost everything you need. Level 1 can be thought of as equivalent to `CRITICAL` messages only---Red Alert, panic panic panic. Level 9, on the other hand, can be thought of as equivalent to `DEBUG` messages and above---give me all the things. And then, of course because [every dial needs to unlock more power](https://youtu.be/KOO5S4vxi0o), you can turn it up to 11, which can be thought of as `TRACE` in many logging paradigms.
+
+Now, that flag only covers what Pulumi's engine is doing. Most of the time, you also need data from the providers you're calling. Each API that Pulumi works with has their own format of diagnostic data, and we need to pass in a flag to ensure those logs bubble up. Why is there a separate flag? Well, some Pulumi providers use a schema from the Terraform community to connect to various systems. Many providers didn't really have create-read-update-delete (CRUD) operations mapped out when they started, which is why the Terraform schema exists in the first place. For those providers, an environment variable is needed to surface the right data in the diagnostics. For non-"bridged" providers (meaning providers that don't use the Terraform schema), you won't need that variable to get set, but you do need a different flag to ask those child processes to bubble up their data. Because there are a lot of moving parts in a Pulumi program, there're a lot of places to get data from, and all of these flags allow you to decide what data you need.
+
+So, to get diagnostic data for "bridged" providers, you need to add in the `TF_LOG` environment variable:
+
+```bash
+TF_LOG=DEBUG pulumi up -v=5
+```
+
+How can you tell if a provider uses the Terraform bridge? There's two ways. First, you can [open up the provider in the Registry](https://registry.pulumi.com/) and check the **Package Details** at the bottom of the API docs landing page. If the Notes section lists the provider as based on a Terraform provider, you know the provider uses the Terraform bridge.
+
+ ![An example provider's Package Details that contains a link to the provider's GitHub repo, the open-source license information for the provider, and a note explaining the provider is based off of a Terraform provider.](./package_details.png "Package Details on a provider")
+
+Second, if you'd rather check from the provider's repository, you can search for `tfgen` in the repository's Makefile, like in [this search of the Makefile for the Aiven provider](https://github.com/pulumi/pulumi-aiven/search?l=Makefile&q=tfgen).
```

Comment:
Not sure we need to recommend this - not practically more useful than the previous section, and rather in the weeds.